dbld/build.manifest

```diff
@@ -28,19 +28,18 @@
 # kafka support needs librdkafka >= 1.0.0, which is only available on ubuntu-bionic
 # on Debian we don't build java and use Python3 _after_ buster
 # libcriterion-dev is available starting with bullseye
-debian-stretch		python2,nocriterion,nojava,nokafka,nomqtt
-debian-buster		python2,nocriterion,nojava,nokafka,nomqtt
+debian-stretch		python3,nocriterion,nojava,nokafka,nomqtt
+debian-buster		python3,nocriterion,nojava,nokafka,nomqtt
```

Comment:
If we end up doing this for 4.0, I think we need to mention this time-consuming manual upgrade steps in the NEWS file.

Reply:
I would argue that we should do it not only for 4.0 but any future version (so even if 4.0 not coming right now), as python2 is dead. Nobody should expect anybody to support it. At some point we have to replace the python3 version of the module. That will be a pain (if any) for the users regardless. The only option I see that would help the transaction to provide a version of python module for both 2 and 3, but at this point that is not a really good possibility.

Also note that syslog-ng project itself officially dropped a support for python2 a little more than a year ago: ttps://github.com/syslog-ng/syslog-ng/releases/tag/syslog-ng-3.31.2

(Agree on a part of describing the possible action required from user.)

Reply:
I know about our announcement. We're using Arch, where this is absolutely normal, but in the Debian/Ubuntu world no one would expect such significant breaking changes with a simple `apt upgrade`.
Doing that with v4.0 will hopefully be understandable.

syslog-ng is following a rolling release model, our version number is purely symbolic. All this justifies breaking things time-to-time, but nonetheless, we were careful not to "break too big".

Anyway, let's document the 2-to-3 transition thoroughly, mention to stretch and buster users how we changed string LogMessage dicts to bytes, etc.

Reply:
As we are planning to implement more detailed types in 4.0 in the python-based drivers, it might be easier for us to wait with this for 4.0 and write one very detailed guide about python in 4.0. It could include the removal of python2, the upgrade steps and info about the new types. I think it could be confusing for some to upgrade from python2 to python3, just to see the upgraded code break again in the next release (4.0). It is basically breaking python in 2 smaller steps vs in 1 larger, and I would _prefer_ the latter.

I agree with Kokan that nobody should expect python2 support nowadays, but as 4.0 is very close now and we have been supporting python2 for so long, I think it does not matter if we wait a couple of months with the python2 removal.

Reply:
We decided to merge this to 4.0 only.